### PR TITLE
chore: Split changes for interface difference in generalizing exportable flow

### PR DESCRIPTION
**Pull request by nidhi-nair**
- [x] Introduced `defaultArtifactId` in `GitApplicationMetadata`
- [x] Moved to using `ArtifactExchangeJson` in exportable interface
- [x] Introduced `artifactType` in ExportingMeta
- [x] Corresponding changes across the board for compatibility

## Summary by CodeRabbit

- **Refactor**
  - Renamed `GitApplicationMetadata` to `GitArtifactMetadata` across multiple components for improved clarity and consistency.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
 @NoArgsConstructor
 @Builder(toBuilder = true)
 public class ExportingMetaDTO {
+    Class<?> artifactType;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java:16]: would this hold the artifact?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 2]
```diff
@@ -3040,10 +3032,10 @@
...
                                 repoSuffix,
-                                gitApplicationMetadata.getRemoteUrl(),
+                                gitArtifactMetadata.getRemoteUrl(),
                                 auth.getPublicKey(),
                                 auth.getPrivateKey(),
-                                gitApplicationMetadata.getBranchName())
+                                gitArtifactMetadata.getBranchName())
                         .map(pushResult -> {
                             if (pushResult.contains("REJECTED")) {
                                 throw new AppsmithException(AppsmithError.GIT_UPSTREAM_CHANGES);
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:3055]: nice catch rabbit!

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java (hunk 1 above)
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java:16]: This would be used to carry the object itself?

**Review by sondermanish — APPROVED** [review]
It's all about `GitArtifactMetadata` folks!!

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ExportingMetaDTO.java:16]: Just the class type, I'm actually changing this to a string value cuz that would be easier to scale. In the next PR :)

**Pull request merged**